main/src/ca/uwaterloo/flix/api/lsp/provider/completion/Completion.scala

```diff
@@ -316,11 +393,41 @@ object Completion {
   /**
     * Represents a keyword completion.
     *
-    * @param name the name of the keyword.
-    * @param priority the completion priority of the keyword.
+    * @param name      the name of the keyword.
+    * @param priority  the completion priority of the keyword.
     */
   case class KeywordCompletion(name: String, priority: Priority) extends Completion
 
+  /**
+    * Represents a keyword literal completion (i.e. `true`).
+    *
+    * The reason we differentiate bewteen normal keywords and these literals
+    * is because completions for the former should include a trailing space
+    * whereas completions for the latter we might not want one.
+    * 
+    * To illustrate this consider the two following correct completions (where ˽ denotes a space)
+    *
+    * `de`      --->    `def˽`
+    * `f(fal)`  --->    `f(false)`
+    *
+    * After the keyword `def` we *always* want a space but if we were
+    * to add a trailing space after `false` we would get the unnatural completion
+    *
+    * `f(fal)`  --->    `f(false˽)`
+    *
+    * @param literal   the literal keyword text.
+    * @param priority  the priority of the keyword.
+    */
+  case class KeywordLiteralCompletion(literal: String, priority: Priority) extends Completion
+
+  /**
+    * Represents a collection keyword completion (i.e. `List#{}`)
+    *
+    * @param keyword   the keyword text (before `#`).
+    * @param priority  the priority of the keyword.
+    */
+  case class KeywordCollectionCompletion(keyword: String, priority: Priority) extends Completion
```

Comment:
I thought we agreed to drop this?

Reply:
yes, we did. But I had a question in regards to this that wanted settled before I did so. Ill copypaste it here:

> I know we came to the conclusion that these couldn't work well, but wouldn't it work fine if they were just suggested above their respective modules? Like, if List#{...} is simply ranked above suggestions to module functions like List.map?

Reply:
But the point is that would be super annoying if I am trying to type List.map

Reply:
Right. I see. Either it'll be at the very bottom or very top, in either of which case it's gonna be annoying in some situation. I'll remove the collection completions :)

Reply:
its gone now